azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/manager/AudioSessionManager.kt

```diff
@@ -32,37 +41,81 @@ internal class AudioSessionManager(
 
     private fun initializeAudioDeviceState() {
-        if (audioManager.isSpeakerphoneOn) {
-            store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.SPEAKER_SELECTED))
-        } else {
-            store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.RECEIVER_SELECTED))
+        when {
+            audioManager.isSpeakerphoneOn ->
+                store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.SPEAKER_SELECTED))
+            audioManager.isBluetoothScoOn ->
+                store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.BLUETOOTH_SCO_SELECTED))
+            else ->
+                store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.RECEIVER_SELECTED))
+
         }
     }
 
     private fun onAudioDeviceStateChange(audioDeviceSelectionStatus: AudioDeviceSelectionStatus) {
         when (audioDeviceSelectionStatus) {
-            AudioDeviceSelectionStatus.SPEAKER_REQUESTED, AudioDeviceSelectionStatus.RECEIVER_REQUESTED ->
+            AudioDeviceSelectionStatus.SPEAKER_REQUESTED, AudioDeviceSelectionStatus.RECEIVER_REQUESTED, AudioDeviceSelectionStatus.BLUETOOTH_SCO_REQUESTED ->
                 switchAudioDevice(audioDeviceSelectionStatus)
         }
     }
 
     private fun switchAudioDevice(audioDeviceSelectionStatus: AudioDeviceSelectionStatus) {
         when (audioDeviceSelectionStatus) {
             AudioDeviceSelectionStatus.SPEAKER_REQUESTED -> {
-                setSpeakerPhoneStatus(true)
+                enableSpeakerPhone()
                 store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.SPEAKER_SELECTED))
             }
             AudioDeviceSelectionStatus.RECEIVER_REQUESTED -> {
-                setSpeakerPhoneStatus(false)
+                enableEarpiece()
                 store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.RECEIVER_SELECTED))
             }
+            AudioDeviceSelectionStatus.BLUETOOTH_SCO_REQUESTED -> {
+                enableBluetooth()
+                store.dispatch(LocalParticipantAction.AudioDeviceChangeSucceeded(AudioDeviceSelectionStatus.BLUETOOTH_SCO_SELECTED))
+            }
         }
     }
 
-    private fun setSpeakerPhoneStatus(status: Boolean) {
-        audioManager.isSpeakerphoneOn = status
+
+
+    private fun enableSpeakerPhone() {
+        audioManager.stopBluetoothSco();
+        audioManager.isBluetoothScoOn = false;
+        audioManager.isSpeakerphoneOn = true;
+    }
+
+    private fun enableEarpiece() {
+        audioManager.stopBluetoothSco();
+        audioManager.isBluetoothScoOn = false;
+        audioManager.isSpeakerphoneOn = false;
+    }
+
+    private fun enableBluetooth() {
+        audioManager.startBluetoothSco();
+        audioManager.isBluetoothScoOn = true;
```

Comment:
just wondering why is `audioManager.isSpeakerphoneOn = false` not needed ?

Reply:
I think a typo or accident, I'll put it back in.

Reply:
good catch, added back